### PR TITLE
Sanketdg/javaextract

### PR DESCRIPTION
**Pull request by SanketDG**

### CODE REVIEW

File: tests/bearlib/languages/documentation/DocumentationCommentTest.py [hunk 1]
```diff
@@ -61,14 +69,6 @@ def check_docstring(self, docstring, expected=[]):
         parsed_metadata = doc_comment.parse()
         self.assertEqual(parsed_metadata, expected)
 
-    def load_testdata(self, filename):
```
Comment (sils1297) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: where is this function used?

Comment (SanketDG) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: [here](https://github.com/coala-analyzer/coala/pull/2390/commits/e2cf672763ddf01e9a00e9199e3ad65c3049065e#diff-603a58b800b18edbad56a01690a21e05R124) 124 and [here](https://github.com/coala-analyzer/coala/pull/2390/commits/e2cf672763ddf01e9a00e9199e3ad65c3049065e#diff-603a58b800b18edbad56a01690a21e05R95) 95

File: tests/bearlib/languages/documentation/DocumentationCommentTest.py [hunk 2]
```diff
@@ -9,6 +9,18 @@
 
 class DocumentationCommentTest(unittest.TestCase):
 
+    Description = DocumentationComment.Description
+    Parameter = DocumentationComment.Parameter
+    ReturnValue = DocumentationComment.ReturnValue
+
+    def load_testdata(self, filename):
+        filename = (os.path.dirname(os.path.realpath(__file__)) +
+                    "/documentation_extraction_testdata/" + filename)
```
Comment (tushar-rishav) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: Hmm shouldn't we use `os.path.join` here? @SanketDG

File: tests/bearlib/languages/documentation/DocumentationCommentTest.py [hunk 3]
```diff
@@ -9,6 +9,18 @@
...
+        with open(filename, "r") as fl:
```
Comment (tushar-rishav) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: @SanketDG In some fonts, the **l** (el) are indistinguishable from the numerals one. Maybe we avoid using them? Maybe it can be made better as:
```python
test_filename = (...)
with open(test_filename,'r') as test_file:
    data = test_file.read()
```

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2390/files/e2cf672763ddf01e9a00e9199e3ad65c3049065e#r68499641

File: coalib/bearlib/languages/documentation/DocumentationComment.py [hunk 4]
```diff
@@ -57,6 +57,9 @@ def parse(self):
         elif self.language == "python" and self.docstyle == "doxygen":
             return self._parse_documentation_with_symbols(
                 ("@param ", " "), "@return ")
+        elif self.language == "java" and self.docstyle == "default":
+            return self._parse_documentation_with_symbols(
+                ("@param  ", " "), "@return ")
```
Comment (sils1297) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: hey, don't we have coalangs from @Makman2 so we don't need if's for every language and style? Hardcoding stuff is bad :(

File: tests/bearlib/languages/documentation/documentation_extraction_testdata/default.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class PhotoGallery {
```
Comment (sils1297) [inline tests/bearlib/languages/documentation/documentation_extraction_testdata/default.java]: is this code copied? License?

**Comment by tushar-rishav** [comment]
unack https://github.com/coala-analyzer/coala/pull/2390/files/e2cf672763ddf01e9a00e9199e3ad65c3049065e#r68500121

File: coalib/bearlib/languages/documentation/DocumentationComment.py (hunk 4 above)
Comment (Makman2) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: that's the next goal, let us first hardcode that, translating this into a coalang system as soon as it gets implemented is easy^^

File: tests/bearlib/languages/documentation/DocumentationCommentTest.py [hunk 6]
```diff
@@ -143,8 +143,9 @@ def test_python_doxygen(self):
...
-        not_implemented = DocumentationComment("some docs", "nolang", "doxygen",
-                                               None, None, None)
-        with self.assertRaises(NotImplementedError):
-            not_implemented.parse()
+
+def test_not_implemented(self):
+    not_implemented = DocumentationComment("some docs", "nolang", "doxygen",
+                                           None, None, None)
+    with self.assertRaises(NotImplementedError):
+        not_implemented.parse()
```
Comment (Makman2) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: why did this went outside a test class?

File: coalib/bearlib/languages/documentation/default.coalang [hunk 7]
```diff
@@ -3,3 +3,7 @@ doc-marker = """, , """
 
 [PYTHON3]
 doc-marker = """, , """
+
+[JAVA]
+doc-marker1 = /**, \ *, \ */
+doc-marker2 = /**, , \ */
```
Comment (Makman2) [inline coalib/bearlib/languages/documentation/default.coalang]: yuo don't use coalangs now, this change should go into an own commit :+1:

File: coalib/bearlib/languages/documentation/DocumentationComment.py (hunk 4 above)
Comment (SanketDG) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: this is my next task

File: coalib/bearlib/languages/documentation/default.coalang (hunk 7 above)
Comment (SanketDG) [inline coalib/bearlib/languages/documentation/default.coalang]: I do use this now :smile:

Comment (SanketDG) [inline coalib/bearlib/languages/documentation/default.coalang]: this is JavaDoc standard

Comment (Makman2) [inline coalib/bearlib/languages/documentation/default.coalang]: but this file is four documentation extraction, not parsing :)

File: tests/bearlib/languages/documentation/DocumentationCommentTest.py (hunk 6 above)
Comment (SanketDG) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: fixed.

File: coalib/bearlib/languages/documentation/default.coalang (hunk 7 above)
Comment (SanketDG) [inline coalib/bearlib/languages/documentation/default.coalang]: and I do use it for extraction?

File: tests/bearlib/languages/documentation/DocumentationCommentTest.py [hunk 8]
```diff
@@ -40,12 +55,14 @@ def test_fields(self):
...
+            "some docs", "nolang", "doxygen", None, None, None)
+        with self.assertRaises(NotImplementedError):
+            not_implemented.parse()
 
-class PythonDocumentationCommentTest(unittest.TestCase):
 
-    Description = DocumentationComment.Description
-    Parameter = DocumentationComment.Parameter
-    ReturnValue = DocumentationComment.ReturnValue
+class PythonDocumentationCommentTest(DocumentationCommentTest):
```
Comment (sils1297) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: so well, if you inherit from that class which has test methods as well, those methods will be inherited and pytest will run all of those twice, right? Can you check that by letting one fail?

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2390/commits/695d1e5f649bb023c868020a17320f607ffb0826

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2390/commits/d545dc645be1a00dc24b2540996f2ca5ac19abf4

File: tests/bearlib/languages/documentation/DocumentationCommentTest.py (hunk 8 above)
Comment (SanketDG) [inline tests/bearlib/languages/documentation/DocumentationCommentTest.py]: you are right I wll fix this

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2390/commits/99fa059f9d4f1d5a8f2b1099f208b5848b5a1222

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2390/commits/fc2e3bfc49b227664fd6dd853d4d00a77c579415

**Comment by sils1297** [comment]
@rultor merge
